core/arch/x86/mangle.c

```diff
@@ -2403,7 +2405,17 @@ mangle_single_step(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr)
      * FIXME i#2144 : to be absolutely transparent, we should translate the
      * exception address as if we did not insert this nop.
```

Comment:
Using the jump target instead of the jump is unique to STATUS_SINGLE_STEP, right?  Nothing else does that.  So the code to do that shouldn't be in general translate.c but instead in callback.c where single step is handled.  IMHO it should be part of this CL: it's just a couple of lines of code and without it the tests fail, don't they?